internal/sdkv2provider/resource_cloudflare_workers_secret_test.go

```diff
@@ -0,0 +1,108 @@
+package sdkv2provider
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+)
+
+const scriptContentForSecret = `addEventListener('fetch', event => {event.respondWith(new Response('test 1'))});`
+
+var workerSecretTestScriptName string
+
+func TestAccCloudflareWorkerSecret_Basic(t *testing.T) {
+	t.Parallel()
+
+	name := generateRandomResourceName()
+	secretText := generateRandomResourceName()
+	workerSecretTestScriptName = generateRandomResourceName()
+	accountId := os.Getenv("CLOUDFLARE_ACCOUNT_ID")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccPreCheckAccount(t)
+		},
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckCloudflareWorkerSecretDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareWorkerSecretWithWorkerScript(workerSecretTestScriptName, name, secretText, accountId),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareWorkerSecretExists(workerSecretTestScriptName, name, accountId),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckCloudflareWorkerSecretDestroy(s *terraform.State) error {
+	accountId := os.Getenv("CLOUDFLARE_ACCOUNT_ID")
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "cloudflare_worker_secret" {
+			continue
+		}
+
+		client := testAccProvider.Meta().(*cloudflare.API)
+		params := cloudflare.ListWorkersSecretsParams{
+			ScriptName: rs.Primary.Attributes["script_name"],
+		}
+
+		secretResponse, err := client.ListWorkersSecrets(context.Background(), cloudflare.AccountIdentifier(accountId), params)
+
+		if err != nil {
+			return err
```

Comment:
This is currently returning an error here because the worker script has been deleted before we can check if the secret was deleted successfully...

Need to delete the secret, do this check, then clean up the rest of the test by deleting the script ideally. Any clues?